eslint-config-base/index.js

```diff
@@ -55,12 +55,94 @@ module.exports = {
       'exports': 'always-multiline',
       'functions': 'only-multiline',
     }],
+    'array-callback-return': [2, { allowImplicit: true }],
+    'complexity': [2, 10],
+    'default-case': [2, { commentPattern: '^no default$' }],
+    'default-case-last': 2,
+    'guard-for-in': 2,
+    'max-classes-per-file': [2, 1], 
+    'no-alert': 2,
+    'no-else-return': [2, { allowElseIf: false }],
+    'no-empty-function': [2, {
+      allow: [
+        'arrowFunctions',
+        'methods'
+      ]
+    }],
+    'no-implicit-coercion': [2, {
+      boolean: false,
+      number: true,
+      string: true,
+      allow: []
+    }],
+    'no-magic-numbers': [2, {
+      ignore: [],
+      ignoreArrayIndexes: true,
+      enforceConst: true,
+      detectObjects: false
+    }],
+    'no-param-reassign': [2, {
+      props: true,
+      ignorePropertyModificationsFor: [
+        'acc', // for reduce accumulators
+        'accumulator', // for reduce accumulators
+        'e', // for e.returnvalue
+        'req', // for Express requests
+        'request', // for Express requests
+        'res', // for Express responses
+        'staticContext' // for ReactRouter context
+      ]
+    }],
+    'no-restricted-properties': [2, {
+      object: 'arguments',
+      property: 'callee',
+      message: 'arguments.callee is deprecated'
+    }, {
+      object: 'global',
+      property: 'isFinite',
+      message: 'Please use Number.isFinite instead'
+    }, {
+      object: 'self',
+      property: 'isFinite',
+      message: 'Please use Number.isFinite instead'
+    }, {
+      object: 'window',
+      property: 'isFinite',
+      message: 'Please use Number.isFinite instead'
+    }, {
+      object: 'global',
+      property: 'isNaN',
+      message: 'Please use Number.isNaN instead'
+    }, {
+      object: 'self',
+      property: 'isNaN',
+      message: 'Please use Number.isNaN instead'
+    }, {
+      object: 'window',
+      property: 'isNaN',
+      message: 'Please use Number.isNaN instead'
+    }, {
+      property: '__defineGetter__',
+      message: 'Please use Object.defineProperty instead.'
+    }, {
+      property: '__defineSetter__',
+      message: 'Please use Object.defineProperty instead.'
```

Comment:
never seen `__defineSetter__` in out codebase. maybe it looks like we deny something that nobody used before and probably don't know :)

Reply:
so maybe I overdid it a bit 😅